cmd/ctr/overlaybd_conv.go

```diff
@@ -749,3 +820,166 @@ func (wc *writeCountWrapper) Write(p []byte) (n int, err error) {
+		})
+	}
+	artifactManifest := artifactspec.Manifest{
+		MediaType:    "application/vnd.cncf.oras.artifact.manifest.v1+json",
+		Blobs:        blobDesc,
+		ArtifactType: artifactTypeName,
+		Subject: artifactspec.Descriptor{
+			MediaType:   srcManifestDesc.MediaType,
+			Digest:      srcManifestDesc.Digest,
+			Size:        srcManifestDesc.Size,
+			URLs:        srcManifestDesc.URLs,
+			Annotations: srcManifestDesc.Annotations,
+		},
+	}
+
+	manifestBytes, err := json.MarshalIndent(artifactManifest, "", "   ")
+	if err != nil {
+		return err
+	}
+	manifestDescriptor := ocispec.Descriptor{
+		MediaType: artifactspec.MediaTypeArtifactManifest,
+		Digest:    digest.FromBytes(manifestBytes),
+		Size:      int64(len(manifestBytes)),
+	}
+
+	refName := remotes.MakeRefKey(ctx, manifestDescriptor)
+
+	// create the oras oci store
+	artifactStore, err := ocitarget.New(contentStoreRootPath)
+	if err != nil {
+		fmt.Println("failed to create ORAS store: Line 872")
+		return err
+	}
+
+	// store artifact manifest in content store
+	// add garbage collection reference labels so blobs are not deleted
+	labels := map[string]string{}
+	labels["containerd.io/gc.root"] = string(manifestDescriptor.Digest)
+	labels["containerd.io/gc.ref.content.config"] = obdManifestConfig.Digest.String()
+	for i, ch := range obdManifest.Layers {
+		labels[fmt.Sprintf("containerd.io/gc.ref.content.l.%d", i)] = ch.Digest.String()
+	}
+	err = content.WriteBlob(ctx, cs, refName, bytes.NewReader(manifestBytes), manifestDescriptor, content.WithLabels(labels))
+	if err != nil {
+		fmt.Printf("failed write artifact manifest blob to content store: %v (Line 886)\n", manifestDescriptor.Digest)
+		return err
+	}
+
+	// tag the manifest with a ref
+	err = artifactStore.Tag(ctx, manifestDescriptor, refName)
+	if err != nil {
+		fmt.Println("failed to tag new artifact manifest: Line 893")
+		return err
+	}
+
+	// create Repository Target
+	repository, err := registry.NewRepository(artifactTarget)
+	if err != nil {
+		fmt.Printf("failed to create new repository for target %s: Line 219\n", artifactTarget)
+		return err
+	}
+
+	// set the Repository Client Credentials
+	repositoryClient := &orasauth.Client{
+		Header: http.Header{
+			"User-Agent": {"overlaybd"},
+		},
+		Cache:      orasauth.DefaultCache,
+		Credential: credentialProvider,
+	}
+
+	// set the TSLClientConfig for HTTP client if insecure set to true
+	if insecure {
+		repositoryClient.Client = http.DefaultClient
+		repositoryClient.Client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+
+	// set the PlainHTTP to true if specified
+	repository.PlainHTTP = plainHTTP
+	repository.Client = repositoryClient
+
+	// copy the artifact manifest to remote Repository
+	returnedManifestDesc, err := oras.Copy(ctx, artifactStore, refName, repository, artifactTarget, oras.DefaultCopyOptions)
+	if err != nil {
+		fmt.Println("failed to push artifact to target: Line 930")
```

Comment:
If you use the containers logger, there's an option to add line no's, 
	"github.com/containerd/containerd/log"

ex file using the logger, https://github.com/akashsinghal/accelerated-container-image/blob/main/pkg/snapshot/overlay.go#L235

Reply:
hmm I switched to the containerd logger but I don't see an option for line no?